internal/otelcollector/config/metric/gateway/receivers.go

```diff
@@ -0,0 +1,46 @@
+package gateway
+
+import (
+	"fmt"
+
+	"github.com/kyma-project/telemetry-manager/internal/otelcollector/config"
+	"github.com/kyma-project/telemetry-manager/internal/otelcollector/ports"
+)
+
+func makeReceiversConfig() Receivers {
+	return Receivers{
+		OTLP: config.OTLPReceiver{
+			Protocols: config.ReceiverProtocols{
+				HTTP: config.Endpoint{
+					Endpoint: fmt.Sprintf("${%s}:%d", config.EnvVarCurrentPodIP, ports.OTLPHTTP),
+				},
+				GRPC: config.Endpoint{
+					Endpoint: fmt.Sprintf("${%s}:%d", config.EnvVarCurrentPodIP, ports.OTLPGRPC),
+				},
+			},
+		},
+	}
+}
```

Comment:
It's a little bit odd to have a function that makes receivers config, which does not call `makeSingletonKymaStatsReceiverCreatorConfig`, defined next to it. Would it be possible to move the annotation check here?

Reply:
We agreed to leave it as it is because the `makeReceiversConfig()` is used only to create the shared receivers for ALL the pipelines. Then in the config builder, we iterate over all the pipelines and add extra receivers/processor needed depending on the spec of the pipelines.